### ISSUE TITLE
adversarial text code does output right thing

### ISSUE THREAD

**Issue by MilesZhao** [tensorflow/models#4648]
### System information
run a toy example in my mac
 

### Describe the problem
My test set has 101 samples (12 classes) at my first cross validation fold. I use 10-cross validation. So I need to collect the predications and corresponding real labels for calculating combined F1-score. 

At the source code of adversarial text, I adjusted two functions as following:
` 
 def eval_graph(self, dataset='test'):

    inputs = _inputs(dataset, pretrain=False)
    
    embedded = self.layers['embedding'](inputs.tokens)
    _, next_state, logits, _ = self.cl_loss_from_embedding(
        embedded, inputs=inputs, return_intermediates=True)

    if FLAGS.single_label:
      indices = tf.stack([tf.range(FLAGS.batch_size), inputs.length - 1], 1)
      labels = tf.expand_dims(tf.gather_nd(inputs.labels, indices), 1)
      weights = tf.expand_dims(tf.gather_nd(inputs.weights, indices), 1)
    else:
      labels = inputs.labels
      weights = inputs.weights
    eval_ops = {
        'accuracy':
            tf.contrib.metrics.streaming_accuracy(
                layers_lib.predictions(logits), labels, weights)
    }

    with tf.control_dependencies([inputs.save_state(next_state)]):
      acc, acc_update = eval_ops['accuracy']
      acc_update = tf.identity(acc_update)
      eval_ops['accuracy'] = (acc, acc_update)

    var_restore_dict = make_restore_average_vars_dict()
    preds_op = layers_lib.predictions(logits)
    actual_op = tf.identity(labels)
    return eval_ops, var_restore_dict, preds_op, actual_op
`

and

`
def run_eval(eval_ops,preds_op,actuals_op, summary_writer, saver):
 
  sv = tf.train.Supervisor(
      logdir=FLAGS.eval_dir, saver=None, summary_op=None, summary_writer=None)
  with sv.managed_session(
      master=FLAGS.master, start_standard_services=False) as sess:
    if not restore_from_checkpoint(sess, saver):
      return
    sv.start_queue_runners(sess)

    metric_names, ops = zip(*eval_ops.items())
    value_ops, update_ops = zip(*ops)

    value_ops_dict = dict(zip(metric_names, value_ops))
    cnt = 0
    # Run update ops
    p = []
    flatten = lambda l: [item for sublist in l for item in sublist]
    num_batches = int(math.ceil(FLAGS.num_examples / FLAGS.batch_size))
    tf.logging.info('Running %d batches for evaluation.', num_batches)
    for i in range(num_batches):
      if (i + 1) % 10 == 0:
        tf.logging.info('Running batch %d/%d...', i + 1, num_batches)
      if (i + 1) % 50 == 0:
        _log_values(sess, value_ops_dict)
      _, preds,actuals =sess.run([update_ops,preds_op,actuals_op])
      cnt = cnt + len(preds)
      
      if i!=26:
        p = p + flatten(actuals)
      else:
        p = p + flatten(actuals[:5])
    print('total number is :', cnt,len(p))
    counter=collections.Counter(p)
    print(counter)
    _log_values(sess, value_ops_dict, summary_writer=summary_writer)
`
It turns out variable 

`actuals` 

does not output the right labels and its numbers like the input.

Input: Counter({5: 30, 1: 20, 11: 14, 3: 8, 6: 7, 2: 7, 10: 4, 8: 3, 4: 3, 0: 2, 9: 2, 7: 1}) 5:30 means class of '5' has 30 samples in test set.
`actual`: Counter({0: 41, 1: 20, 11: 14, 3: 8, 6: 7, 10: 4, 8: 3, 9: 2, 7: 1, 4: 1}). Miss two classes and the numbers also partially incorrect. 

I suspect the function of `tf.contrib.training.batch_sequences_with_states` messes around my code. But I do not know how to fix it.

One more thing is my batch size is 32 and there are 101 examples. SO the inputs will be exhausted at last batch  (only 5 sample left). I found that I still can get 32 samples at final batch. When I count it.

Thanks.

code link https://github.com/tensorflow/models/tree/master/research/adversarial_text

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
